### DISCUSSION TITLE
The Uno Platform version of Community Toolkit is not supported on WinAppSDK

### DISCUSSION THREAD

**Discussion by DJDQfff** [unoplatform/uno#13152] [Q&A]
# Configuration
 * Version:4.9.45
* targetplatfomr: android , windows(WinUI)
* .net 7
# Trouble :
    Showed by InteliSense: only x:Name and InitialComponent() . problem.
# What I did 
    I tried to sovle the problem as https://platform.uno/docs/articles/get-started-wizard.html?tabs=windows#common-issues said.
 But It doesn't work.

    What  build error shows  is these:
* Unable to find content value on Uno.UI.SourceGenerators.XamlGenerator.XamlObjectDefinition	.but don't say where the source error is .
* An error was found in Grid  ,  An error was found in Page . Similar errors are showed for multiple xaml files.

And other problem that are the result of them.

**Comment by limeniye** [comment]
It looks like a following problem: https://github.com/unoplatform/uno/issues/12862

Update all your packages from 4.9.45 to 4.10.0-dev***
For example:

**Directory.Build.props**
```XML
    <PropertyGroup>
        <UnoVersion>4.10.0-dev.491</UnoVersion>
    </PropertyGroup>  
```
**Directory.Packages.props**
```XML
  <ItemGroup>
    <PackageVersion Include="Uno.WinUI" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.RemoteControl" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.Skia.Gtk" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.Skia.Linux.FrameBuffer" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.Skia.Wpf" Version="$(UnoVersion)" />
    <PackageVersion Include="Uno.WinUI.WebAssembly" Version="$(UnoVersion)" />
  </ItemGroup>    
```

If you will see **RuntimeIdentifiers error** message add the following lines to your project:
```XML
    <PropertyGroup Condition="'$(TargetFramework)' == 'net7.0-windows10.0.19041'">
        <RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
    </PropertyGroup>
```

For CommunityToolkit use the [following approach](https://platform.uno/docs/articles/uno-community-toolkit.html?tabs=tabid-winui):
```XML
    <ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
        <PackageReference Include="CommunityToolkit.WinUI.UI.Controls" />
    </ItemGroup>
    <ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
        <PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls" />
    </ItemGroup>
```

If after the following steps you will see the WindowsAppSDK version error -- update your SDK packages:
```XML
    <PackageVersion Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
    <PackageVersion Include="Microsoft.WindowsAppSDK" Version="1.3.230602002" />
```

**Rebuild your project twice.**

**Reply by jeromelaban** [reply]
You may want to try the sample DataGrid sample here: https://github.com/unoplatform/Uno.Samples/tree/master/UI/UnoWCTDataGridSample. There are other samples for community toolkit as well in the same folder.
